### PR TITLE
Data for database

### PR DESCRIPTION
**Pull request by JakobJFL**

### CODE REVIEW

File: data_api/insert.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        'title': art.headline,
+        'publication': art.publication,
+        'path': art.path,
+        'date': art.published_at,
+        'author': art.author_name,
+        'totalWords': art.total_words,
+        'uniqueWords': len(art.tokens)
+    }]
+    if art.publisher != 'Nordjyske Medier':
+        print_warning('Article is skipped. Not form Nordjyske Medier. Data: ' + str(json_data))
```
Comment (k-nielsen) [inline data_api/insert.py:28]: Spelling mistake: form -> from

File: data_api/insert.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    if len(art.sub_head) == 0:
+        json_data = [{
+            'documentId': art.id,
+            "index": 0,
+            'content': art.body_text[0]
+        }]
+        make_post(API_URL+'document-contents', json_data)
+    else:
+        json_data = []
+        for index, combination in enumerate(zip(art.sub_head, art.body_text)):
```
Comment (k-nielsen) [inline data_api/insert.py:43]: Is the name "combination" too general possibly?

**Review by k-nielsen — APPROVED** [review]

File: tests/extract_words_test.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        art_input.sub_head = ['test2']
+        art_input.body_text = ['test3']
+        tokens = TokenDict()
+        tokens.add('test1', 1, 1)
+        tokens.add('test2', 1, 2)
+        tokens.add('test3', 1, 3)
+        art_output = add_tokens_to_articles([art_input])
+        self.assertDictEqual(art_output[0].tokens, tokens)
+
+    def test_add_tokens_to_articles_4(self):
```
Comment (Flaiizor) [inline tests/extract_words_test.py:42]: Naming is confusing - maybe name something else than 1, 2, 3, 4?

**Review by k-nielsen — APPROVED** [review]

File: tests/validation_test.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    def test_is_valid_valid1(self):
+        json_data = """{"headline":"text", "paragraphs":[
+            {"kind": "paragraph", "value": "test"},
+            {"kind": "paragraph", "value": "test2"}],
+            "publication":"pub", "byline": {"name": "k", "email": "k@dk.dk"},
+            "published_at": "t", "publisher": "Nordjyske Medier","publication": "h"}"""
+        data = json.loads(json_data)
+        self.assertTrue(is_valid(data))
+
+    def test_is_valid_valid2(self):
```
Comment (Flaiizor) [inline tests/validation_test.py:26]: valid1 and valid2 - Patrick must make better names :)

File: tests/extract_words_test.py (hunk 3 above)
Comment (k-nielsen) [reply tests/extract_words_test.py:42]: yes

File: tests/gen_vec_words_test.py [hunk 5]
```diff
@@ -1,35 +1,33 @@
...
+                      ('   ', []), ('32432', []), ('cat234', ['cat'])]
         for p_1, p_2 in param_list:
             with self.subTest():
-                self.assertEqual(make_unique_words(p_1), p_2)
+                self.assertEqual(get_only_words(p_1), p_2)
 
 
 class TestGetUniqueWords(unittest.TestCase):
-    def test_get_unique_words_params(self):
+    def test_get_words_from_articles_1(self):
```
Comment (Flaiizor) [inline tests/gen_vec_words_test.py:17]: @Patrickggwpez make name good thank you

**Review by Flaiizor — COMMENTED** [review]
All in all, very nice. Fix some naming, then all is good.

**Pull request merged**